extensions/api/data-management/contractdefinition/src/main/java/org/eclipse/dataspaceconnector/api/datamanagement/contractdefinition/model/ContractDefinitionDto.java

```diff
@@ -0,0 +1,104 @@
+/*
+ *  Copyright (c) 2020 - 2022 Microsoft Corporation
+ *
+ *  This program and the accompanying materials are made available under the
+ *  terms of the Apache License, Version 2.0 which is available at
+ *  https://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  SPDX-License-Identifier: Apache-2.0
+ *
+ *  Contributors:
+ *       Microsoft Corporation - initial API and implementation
+ *
+ */
+
+package org.eclipse.dataspaceconnector.api.datamanagement.contractdefinition.model;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
+import com.fasterxml.jackson.databind.annotation.JsonPOJOBuilder;
+import org.eclipse.dataspaceconnector.spi.query.Criterion;
+
+import java.util.List;
+
+@JsonDeserialize(builder = ContractDefinitionDto.Builder.class)
+public class ContractDefinitionDto {
+    private String accessPolicyId;
+    private String contractPolicyId;
+    private List<Criterion> criteria;
+    private String id;
+
+    private ContractDefinitionDto() {
+    }
+
+    public String getAccessPolicyId() {
+        return accessPolicyId;
+    }
+
+    public void setAccessPolicyId(String accessPolicyId) {
+        this.accessPolicyId = accessPolicyId;
+    }
+
+    public String getContractPolicyId() {
+        return contractPolicyId;
+    }
+
+    public void setContractPolicyId(String contractPolicyId) {
+        this.contractPolicyId = contractPolicyId;
+    }
+
+    public List<Criterion> getCriteria() {
+        return criteria;
+    }
+
+    public void setCriteria(List<Criterion> criteria) {
+        this.criteria = criteria;
+    }
+
+    public String getId() {
+        return id;
+    }
+
+    public void setId(String id) {
+        this.id = id;
+    }
+
+
+    @JsonPOJOBuilder(withPrefix = "")
+    public static final class Builder {
+        private final ContractDefinitionDto dto;
+
+        private Builder() {
+            dto = new ContractDefinitionDto();
+        }
+
+        @JsonCreator
+        public static Builder newInstance() {
+            return new Builder();
+        }
+
+        public Builder accessPolicyId(String accessPolicyId) {
+            dto.accessPolicyId = accessPolicyId;
+            return this;
+        }
+
+        public Builder contractPolicyId(String contractPolicyId) {
+            dto.contractPolicyId = contractPolicyId;
+            return this;
+        }
+
+        public Builder criteria(List<Criterion> criteria) {
+            dto.criteria = criteria;
+            return this;
+        }
+
+        public Builder id(String id) {
+            dto.id = id;
+            return this;
+        }
```

Comment:
Is there a reason why you set the attributes directly and not use the setter methods?

Reply:
we would have to have (private) setter methods only for that reason, which makes little sense.
There shouldn't be public setters (unless there are good and specific reasons), since we perform object creation with Builders.

Reply:
... and I just saw that there are in fact setter :)
IntelliJ must have shown preemptive obedience. fixed.

Reply:
Okaay. But you have public setters (e.g., lines 38-40). Or are you and I talking about different things?
What I meant was this:

```java
        public Builder accessPolicyId(String accessPolicyId) {
            dto.setAccessPolicyId (accessPolicyId);
            return this;
        }
```

Reply:
seems like a comment race condition. check my comment above

Reply:
anyway, thanks for noticing!

Reply:
All right, then I'm happy and quiet ;-) Thanks for the explanation!